extension/extensionlimiter/weight.go

```diff
@@ -0,0 +1,82 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package extensionlimiter // import "go.opentelemetry.io/collector/extension/extensionlimiter"
+
+import "slices" // WeightKey is an enum type for common rate limits.  The
+// StandardAllKeys, StandardMiddlewareKeys, and
+// StandardNotMiddlewareKeys methods return the list of middleware
+// keys that can be automatically configured through middleware and
+// not.
+type WeightKey string
```

Comment:
Not sure how important this is, do you see a big usage of this on the critical path? If yes, should we use an enum as int instead, since that will be a bit faster for type checks?

Reply:
My expectation is that weight keys are used during Start() where the providers are called to initialize the actual limiter instances. We have discussed some ideas about new things that could be rate limited, and I believe @axw previously suggested this might be an open set, not a closed one; either way the goal should be to bind limiters once via a provider and use them w/o passing around weight keys at runtime.